File: stepik-union/src/main/java/com/jetbrains/tmp/learning/courseFormat/StudyStatus.java
package com.jetbrains.tmp.learning.courseFormat;

import org.jetbrains.annotations.Nullable;

public enum StudyStatus {
    UNCHECKED, SOLVED, FAILED;

    public static StudyStatus of(@Nullable String status) {
        if (status == null) {
            return UNCHECKED;
        }

        status = status.toLowerCase();

        if (status.equals("correct") || status.equals("solved")) {
            return SOLVED;
        }

        if (status.equals("wrong") || status.equals("failed")) {
            return FAILED;
        }

        return UNCHECKED;
    }
}

File: stepik-core/src/main/kotlin/org/stepik/core/courseFormat/StudyStatus.kt
package org.stepik.core.courseFormat

enum class StudyStatus {
    UNCHECKED,
    FAILED,
    SOLVED;


    companion object {

        fun of(status: String?): StudyStatus {
            return when (status?.toLowerCase()) {
                "correct", "solved" -> SOLVED
                "wrong", "failed" -> FAILED
                else -> UNCHECKED
            }
        }
    }
}
